scenario/constraints/CometBalanceConstraint.ts

```diff
@@ -75,12 +75,12 @@ export class CometBalanceConstraint<T extends CometContext, R extends Requiremen
       // XXX ideally when properties fail
       //  we can report the names of the solution which were applied
       const solutions = [];
-      solutions.push(async function barelyMeet(context: T, world: World) {
+      solutions.push(async function barelyMeet(context: T) {
```

Comment:
As much as I like having a world, I do think this makes life so much easier and doesn't really change semantics.

Reply:
its still being passed, fwiw

Reply:
Like scen plugin doesn't assume that it will be in the context, but it *is* for comet, which I think was mainly what you wanted to preserve, maybe?

Reply:
The idea of world is that you could encapsulate side-effects in an object, but the side-effects mostly live in the chain state, wherein the World isn't as useful as some pure thing in a mostly impure world.